remixer_storage/src/main/java/com/google/android/libraries/remixer/storage/FirebaseRemoteControllerSyncer.java

```diff
@@ -212,4 +255,43 @@ public void onChildMoved(DataSnapshot dataSnapshot, String childKey) {
   public void onCancelled(DatabaseError databaseError) {
 
   }
+
+  public void addSharingStatusListener(@NonNull SharingStatusListener listener) {
+    listener.updateSharingStatus(sharing);
+    listeners.add(new WeakReference<>(listener));
+  }
+
+  public Intent getShareLinkIntent() {
+    Intent intent = new Intent();
+    intent.setAction(Intent.ACTION_SEND);
+    String url =String.format(
+        Locale.getDefault(),
+        "https://%s.firebaseapp.com/%s",
+        getUrlPrefix(),
+        remoteId);
+    String message =
+        applicationContext.getResources().getString(R.string.shareLinkFormat, url);
+    intent.putExtra(Intent.EXTRA_TEXT, message);
+    intent.setType("text/plain");
+    return intent;
+  }
+
+  private String getUrlPrefix() {
+    String url = FirebaseApp.getInstance().getOptions().getStorageBucket();
+    if (url.contains(".")) {
+      return url.substring(0, url.indexOf('.'));
+    }
+    return url;
+  }
+
+  /**
+   * Implementors are objects who are listening to changes to whether we stop/start sharing.
```

Comment:
Slightly confused by this - is this saying that the objects that implement SharingStatusListener listen to changes in sharing status? Or that this interface itself listens to changes?